### PR TITLE
Change behavior when handling an OFPBAC_BAD_OUT_PORT error message

### PR DESCRIPTION
**Pull request by gleybersonandrade**
This PR changes the behavior when handling an OFPBAC_BAD_OUT_PORT error message, checking the interface config with OFPPC_NO_FWD flag.

Fix https://github.com/kytos/of_core/issues/80

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -163,9 +167,33 @@ def handle_errors(self, event):
...
+        error_packet = connection.protocol.unpack(error_data)
+
+        if (message.code == BadActionCode01.OFPBAC_BAD_OUT_PORT or
+           message.code == BadActionCode04.OFPBAC_BAD_OUT_PORT):
+
+            for action in error_packet.actions:
+                try:
+                    iface = switch.get_interface_by_port_no(action.port.value)
+                except AttributeError:
+                    iface = switch.get_interface_by_port_no(action.port)
```
Comment (hdiogenes) [inline main.py:185]: I feel like there's something wrong going on here. `action.port` can act the same as `action.port.value`? Without diving into the details, it looks like `action.port` should have a setter (maybe defined as a Python `property`) and try to be always the same kind of object.

File: main.py [hunk 2]
```diff
@@ -2,6 +2,10 @@
 from collections import OrderedDict
 
 from flask import jsonify, request
+from pyof.v0x01.asynchronous.error_msg import BadActionCode as BadActionCode01
+from pyof.v0x01.common.phy_port import PortConfig as PortConfig01
+from pyof.v0x04.asynchronous.error_msg import BadActionCode as BadActionCode04
+from pyof.v0x04.common.port import PortConfig as PortConfig04
```
Comment (hdiogenes) [inline main.py:8]: To me, these imports are too specific to be here. More on that below.

File: main.py [hunk 3]
```diff
@@ -163,9 +167,33 @@ def handle_errors(self, event):
...
+        message = event.content["message"]
+        connection = event.source
+        switch = connection.switch
+
+        error_data = message.data.pack()
+        # Get the packet responsible for the error
+        error_packet = connection.protocol.unpack(error_data)
+
+        if (message.code == BadActionCode01.OFPBAC_BAD_OUT_PORT or
+           message.code == BadActionCode04.OFPBAC_BAD_OUT_PORT):
```
Comment (hdiogenes) [inline main.py:179]: Try to think this way: when we implement OpenFlow 1.4 in `python-openflow`, will we have to come back here and also add support for it? Could we try to make this more "future-proof"?

File: main.py [hunk 4]
```diff
@@ -163,9 +167,33 @@ def handle_errors(self, event):
...
+                    iface = switch.get_interface_by_port_no(action.port.value)
+                except AttributeError:
+                    iface = switch.get_interface_by_port_no(action.port)
+
+                # Check interface to drop packets forwarded to it
+                if iface:
+                    if connection.protocol.version == 0x01:
+                        iface.config = PortConfig01.OFPPC_NO_FWD
+                    elif connection.protocol.version == 0x04:
+                        iface.config = PortConfig04.OFPPC_NO_FWD
```
Comment (hdiogenes) [inline main.py:192]: Same thing here. There should be a function somewhere that returned the right object for any `pyof` supported version.

**Review by hdiogenes — CHANGES_REQUESTED** [review]

File: main.py (hunk 4 above)
Comment (hdiogenes) [reply main.py:192]: Something like this, just to have an idea of what I'm talking about:

```
port_config = pyof.get_port_config_for_version(connection.protocol.version)
iface.config = port_config.OFPPC_NO_FWD
```

**Comment by josemauro** [comment]
The changes in this PR were moved and improved in PR #92.

**Pull request closed without merging**